core/src/elo-connector/elo-connector.service.ts

```diff
@@ -0,0 +1,291 @@
+import {
+    InjectQueue, OnGlobalQueueCompleted, Processor,
+} from "@nestjs/bull";
+import {Injectable, Logger} from "@nestjs/common";
+import {InjectRepository} from "@nestjs/typeorm";
+import type {BallchasingPlayer} from "@sprocketbot/common";
+import {Job, Queue} from "bull";
+import {Repository} from "typeorm";
+
+import type {
+    MatchParent, PlayerStatLine,
+} from "../database";
+import {
+    Match,
+    Player, Round, Team,
+} from "../database";
+import type {
+    MatchSummary, PlayerSummary, SalaryPayloadItem, SeriesStatsPayload,
+} from "./elo.types";
+import {
+    GameMode, SeriesType, TeamColor,
+} from "./elo.types";
+
+@Injectable()
+@Processor("elo")
+export class EloConnectorService {
+    private readonly logger = new Logger(EloConnectorService.name);
+
+    constructor(
+        @InjectQueue("elo") private eloQueue: Queue,
+        @InjectRepository(Player) private readonly playerRepository: Repository<Player>,
+        @InjectRepository(Round) private readonly roundRepository: Repository<Round>,
+        @InjectRepository(Team) private readonly teamRepository: Repository<Team>,
+        @InjectRepository(Match) private readonly matchRepository: Repository<Match>,
+    ) { }
+
+    /* eslint-disable */
```

Comment:
Why disable everything?

Reply:
It is related to the `result: any` problem, and that's one that's out of my hands as far as I can tell. Bull doesn't provide type information for job return types, since they're so arbitrarily defined as JSON objects in redis. I do think that maybe we can ZOD both sides of the pipe as we do elsewhere and lock this down a bit, but it needs more thought than I've been able to spare on this the last few weeks.

Reply:
Probably the big piece we're missing here that's causing this is ELO's publishing of a types package, which we've discussed and decided upon as a design, but I just haven't gotten around to yet.

Reply:
> Bull doesn't provide type information for job return types, since they're so arbitrarily defined as JSON objects in redis.  

This is the big use case for zod; however, as you said:  

> the big piece we're missing here that's causing this is ELO's publishing of a types package  

I've logged a ticket; we'll come back around to this, assuming we don't have any major issues

Reply:
https://app.clickup.com/t/2pdz0e8